src/components/Search/SearchAPI.js

```diff
@@ -25,32 +25,102 @@ export const fetchAggregatedItems = async (
     throw new Error(`An error has occurred: ${response.status}`)
   }
   const apiResponse = await response.json()
-  const apikeys = apiResponse.aggregations?.find((el) => el.name === frequency)
+  const apiKeys = apiResponse.aggregations?.find(
+    (el) => el.name === frequency
+  ).buckets
 
   const numberMatched = apiResponse?.aggregations?.find(
     (el) => el.name === 'total_count'
   )?.value
 
   // create Geojson file with matched geometry and frequency
-  const mappedKeys = apikeys.buckets
-    .map((value) => {
-      const apiValue = value.key
-      const gridKey = apiValue.split('-')
-      const coordinates =
-        _gridCellData[gridKey[0].toLowerCase()].cells[gridKey[1]]
+  const mappedKeysToGrid = apiKeys
+    .map((feature) => {
+      let coordinates =
+        _gridCellData[feature.key.split('-')[0].toLowerCase()].cells[
+          feature.key.split('-')[1]
+        ]
+
+      // for multipolygon grid cells, extract the next array
+      if (coordinates && coordinates[0].length === 1) {
+        coordinates = coordinates[0]
+      }
       return {
         geometry: {
-          type: _gridCellData[gridKey[0].toLowerCase()].type,
+          type: 'Polygon',
           coordinates
         },
         type: 'Feature',
         properties: {
-          'grid:code': value.key,
-          frequency: value.frequency
+          'grid:code': feature.key,
+          frequency: feature.frequency
         }
       }
     })
-    .filter((x) => x)
+    .filter((feature) => feature.geometry.coordinates)
+
+  return {
+    type: 'FeatureCollection',
+    features: mappedKeysToGrid,
+    numberMatched,
+    searchType: 'AggregatedResults'
+  }
+}
+
+export const fetchGeoHexItems = async (searchParamsStr, zoomLevel) => {
+  const precision = Math.round(zoomLevel / 3)
 
-  return { type: 'AggregatedResults', features: mappedKeys, numberMatched }
+  // fetch frequency and counts from API
+  const searchURL = `${process.env.REACT_APP_STAC_API_URL}/aggregate?${searchParamsStr}&aggregations=grid_geohex_frequency,total_count&grid_geohex_frequency_precision=${precision}`
+  const response = await fetch(searchURL)
+  if (!response.ok) {
+    throw new Error(`An error has occurred: ${response.status}`)
+  }
+  const apiResponse = await response.json()
+  const buckets = apiResponse.aggregations?.find(
+    (el) => el.name === 'grid_geohex_frequency'
+  ).buckets
+
+  const numberMatched = apiResponse?.aggregations?.find(
+    (el) => el.name === 'total_count'
+  )?.value
+
+  // set up the max and min values for frequency results
+  const frequencyArr = buckets.map((feature) => feature.frequency)
+  const frequencyMax = Math.max.apply(null, frequencyArr)
+  const frequencyMin = Math.min.apply(null, frequencyArr)
+
+  const h3 = require('h3-js')
+
+  const convertedItems = buckets.map((feature) => {
+    const hexBoundary = h3.cellToBoundary(feature.key)
```

Comment:
```suggestion
    const hexBoundary = h3.cellToBoundary(feature.key, true)
```

this returns the boundaries in geojson lon/lat order and "closes the loop" by having the start and end vertex be the same (https://github.com/uber/h3-js#h3celltoboundaryh3index-formatasgeojson--arraycoordpair)

Reply:
This has been resolved.